handler/src/main/java/io/netty/handler/ssl/util/FingerprintTrustManagerFactory.java

```diff
@@ -207,4 +247,52 @@ protected void engineInit(ManagerFactoryParameters managerFactoryParameters) thr
     protected TrustManager[] engineGetTrustManagers() {
         return new TrustManager[] { tm };
     }
+
+    /**
+     * A builder for creating {@link FingerprintTrustManagerFactory}.
+     */
+    public static final class Builder {
+
+        /**
+         * A hash algorithm for fingerprints.
+         */
+        private final String algorithm;
+
+        /**
+         * A list of fingerprints.
+         */
+        private final List<String> fingerprints = new ArrayList<String>();
+
+        /**
+         * Creates a builder.
+         *
+         * @param algorithm a hash algorithm
+         */
+        private Builder(String algorithm) {
+            this.algorithm = ObjectUtil.checkNotNull(algorithm, "algorithm");
+        }
+
+        /**
+         * Adds fingerprints.
+         *
+         * @param fingerprints a number of fingerprints
+         * @return the same builder
+         */
+        public Builder fingerprints(String... fingerprints) {
+            ObjectUtil.checkNotNull(fingerprints, "fingerprints");
+            for (String fingerprint : fingerprints) {
+                this.fingerprints.add(fingerprint);
+            }
+            return this;
+        }
+
+        /**
+         * Creates a {@link FingerprintTrustManagerFactory}.
+         *
+         * @return a new {@link FingerprintTrustManagerFactory}
+         */
+        public FingerprintTrustManagerFactory build() {
+            return new FingerprintTrustManagerFactory(this.algorithm, toFingerprintArray(this.fingerprints));
```

Comment:
What would be the expected behavior when `fingerprints` is empty? Throwing an `IllegalStateException` in `build()`?

Reply:
Currently, no fingerprints are allowed. That would mean that no certificate is going to be trusted. In theory, it's okay but it looks unlikely that an application may want to use such a paranoid `TrustManager`. Therefore, I think it would be fine to throw an exception if no fingerprints are provided. I'd put this check in to a constructor since all the checks are currently done there, and `FingerprintTrustManagerFactory` may be still created by calling the deprecated constructors.